### ISSUE TITLE
Kilosort spike sorting failed cannot compute length

### ISSUE THREAD

**Issue by AbhiSwamiUConn** [MouseLand/Kilosort#712]
### Describe the issue:

Hi, I am currently using spikeinterface to run kilosort version 4.0.6 on a plexon file in spike interface.  This error occurs when I try sorting the file after running a bandpass filter and attaching a generated multishank probe using probeinterface.  
```
C:\Users\User\anaconda3\Lib\site-packages\threadpoolctl.py:1214: RuntimeWarning: 
Found Intel OpenMP ('libiomp') and LLVM OpenMP ('libomp') loaded at
the same time. Both libraries are known to be incompatible and this
can cause random crashes or deadlocks on Linux when loaded in the
same Python program.
Using threadpoolctl may cause crashes or deadlocks. For more
information and possible workarounds, please see
    https://github.com/joblib/threadpoolctl/blob/master/multiple_openmp.md

  warnings.warn(msg, RuntimeWarning)
C:\Users\User\anaconda3\Lib\site-packages\numpy\core\fromnumeric.py:3464: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
C:\Users\User\anaconda3\Lib\site-packages\numpy\core\_methods.py:192: RuntimeWarning: invalid value encountered in divide
  ret = ret.dtype.type(ret / rcount)
---------------------------------------------------------------------------
SpikeSortingError                         Traceback (most recent call last)
Cell In[5], line 2
      1 #Sort
----> 2 sorting = ss.run_sorter("kilosort4", recording_f, output_folder= output_folder1, remove_existing_folder= True)
      3 Sanalyzer = si.waveform_extractor.extract_waveforms(recording=recording_f, sorting=sorting, folder = output_folder2, overwrite= True)

File ~\anaconda3\Lib\site-packages\spikeinterface\sorters\runsorter.py:175, in run_sorter(sorter_name, recording, output_folder, remove_existing_folder, delete_output_folder, verbose, raise_error, docker_image, singularity_image, delete_container_files, with_output, **sorter_params)
    168             container_image = singularity_image
    169     return run_sorter_container(
    170         container_image=container_image,
    171         mode=mode,
    172         **common_kwargs,
    173     )
--> 175 return run_sorter_local(**common_kwargs)

File ~\anaconda3\Lib\site-packages\spikeinterface\sorters\runsorter.py:225, in run_sorter_local(sorter_name, recording, output_folder, remove_existing_folder, delete_output_folder, verbose, raise_error, with_output, **sorter_params)
    223 SorterClass.set_params_to_folder(recording, output_folder, sorter_params, verbose)
    224 SorterClass.setup_recording(recording, output_folder, verbose=verbose)
--> 225 SorterClass.run_from_folder(output_folder, raise_error, verbose)
    226 if with_output:
    227     sorting = SorterClass.get_result_from_folder(output_folder, register_recording=True, sorting_info=True)

File ~\anaconda3\Lib\site-packages\spikeinterface\sorters\basesorter.py:293, in BaseSorter.run_from_folder(cls, output_folder, raise_error, verbose)
    290         print(f"{sorter_name} run time {run_time:0.2f}s")
    292 if has_error and raise_error:
--> 293     raise SpikeSortingError(
    294         f"Spike sorting error trace:\n{log['error_trace']}\n"
    295         f"Spike sorting failed. You can inspect the runtime trace in {output_folder}/spikeinterface_log.json."
    296     )
    298 return run_time

SpikeSortingError: Spike sorting error trace:
Traceback (most recent call last):
  File "C:\Users\User\anaconda3\Lib\site-packages\spikeinterface\sorters\basesorter.py", line 258, in run_from_folder
    SorterClass._run_from_folder(sorter_output_folder, sorter_params, verbose)
  File "C:\Users\User\anaconda3\Lib\site-packages\spikeinterface\sorters\external\kilosort4.py", line 238, in _run_from_folder
    ops, bfile, st0 = compute_drift_correction(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\kilosort\run_kilosort.py", line 350, in compute_drift_correction
    ops, st = datashift.run(ops, bfile, device=device, progress_bar=progress_bar)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\kilosort\datashift.py", line 192, in run
    st, _, ops  = spikedetect.run(ops, bfile, device=device, progress_bar=progress_bar)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\kilosort\spikedetect.py", line 198, in run
    ops = template_centers(ops)
          ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\User\anaconda3\Lib\site-packages\kilosort\spikedetect.py", line 95, in template_centers
    ops['yup'] = np.arange(ymin, ymax+.00001, dmin/2)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: arange: cannot compute length

Spike sorting failed. You can inspect the runtime trace in C:\Users\User\Desktop\files1/spikeinterface_log.json.
```

**Comment by jacobpennington** [comment]
This looks like your probe file is not generated correctly. You would get that error if ymin = ymax, i.e. the smallest and largest y-value for the probe are the same.

**Comment by AbhiSwamiUConn** [comment]
Sorry for the late reply.  I do not understand what could be causing the probe file to not be generated properly.  Here is what my probe is displayed as when I try to plot it.  This probe seems to work with the built in sorters for spike interface.  
![image](https://github.com/MouseLand/Kilosort/assets/159394395/5c9656be-1bb1-46ad-afc7-672873397da6)

**Comment by jacobpennington** [comment]
Ah, I see. We just haven't had anyone try to use a probe with only a single channel per shank before. I'll update the code to allow for that, but it might not be updated on PyPi for a day or two since I'm working on adding some other things for the next version update.

**Comment by jacobpennington** [comment]
The fix is live now on PyPi with `v4.0.12`, so I'm closing this. If you still encounter errors after updating to the latest version, please let us know.
